searchlib/src/vespa/searchlib/attribute/singleenumattribute.h

```diff
@@ -110,6 +110,9 @@ class SingleValueEnumAttribute : public B, public SingleValueEnumAttributeBase {
     void onUpdateStat() override;
     void reclaim_memory(generation_t oldest_used_gen) override;
     void before_inc_generation(generation_t current_gen) override;
+    bool isUndefined(DocId doc) const override {
+        return acquire_enum_entry_ref(doc) == this->getEnumStore().get_default_value_ref().load_relaxed();
```

Comment:
Note that during enum store compaction, isUndefined() might return false when it should return true, due to acquire_enum_entry_ref(doc) returning an EntryRef referencing the moved default value, while get_default_value_ref() returns an EntryRef referencing the original default value.

Reply:
hmm, that is probably something we do not want to allow.
Lets discuss options tomorrow.
Could the default_value_ref be fixed for the application lifetime ?